pilot/pkg/networking/core/v1alpha3/route/route_test.go

```diff
@@ -1238,12 +1241,53 @@ func TestBuildHTTPRoutes(t *testing.T) {
 			wildcardIndex,
 		)
 		log.Printf("%#v", vhosts)
-		g.Expect(vhosts).To(HaveLen(2))
+		g.Expect(vhosts).To(HaveLen(2)) // *.example.org, *.hello.example.org
 		for _, vhost := range vhosts {
 			g.Expect(vhost.Services).To(HaveLen(1))
 			g.Expect(vhost.Routes).To(HaveLen(1))
 		}
 	})
+
+	t.Run("for virtualservices with with wildcard hosts outside of the serviceregistry (on port 80)", func(t *testing.T) {
+		g := NewWithT(t)
+		cg := v1alpha3.NewConfigGenTest(t, v1alpha3.TestOptions{
+			Configs: []config.Config{
+				virtualServiceWithWildcardHost,
+				virtualServiceWithNestedWildcardHost,
+				virtualServiceWithGoogleWildcardHost,
+			},
+			Services: []*model.Service{exampleWildcardService, exampleNestedWildcardService},
+		})
+
+		// Redefine the service registry for this test
+		serviceRegistry := map[host.Name]*model.Service{
+			"*.example.org":             exampleWildcardService,
+			"goodbye.hello.example.org": exampleNestedWildcardService,
+		}
+
+		// note that this virtualService doesn't have an entry in the wildcard index
+		wildcardIndex := map[host.Name]types.NamespacedName{
+			"*.example.org":       virtualServiceWithWildcardHost.NamespacedName(),
+			"*.hello.example.org": virtualServiceWithNestedWildcardHost.NamespacedName(),
```

Comment:
why not `*.google.com` included

Reply:
Because only hosts in the serviceRegistry are in the index: https://github.com/istio/istio/blob/1fa86e400ba7dc6f6a13ba54c047323f05d494bc/pilot/pkg/model/sidecar.go#L857

Reply:
IC